rllib/evaluation/collectors/simple_list_collector.py

```diff
@@ -45,24 +49,30 @@ class _AgentCollector:
 
     _next_unroll_id = 0  # disambiguates unrolls within a single episode
 
-    def __init__(self, shift_before: int = 0):
-        self.shift_before = max(shift_before, 1)
+    def __init__(self, view_reqs):
+        self.shift_before = -min(
+            [(int(vr.data_rel_pos.split(":")[0])
+              if isinstance(vr.data_rel_pos, str) else vr.data_rel_pos) +
+             (-1 if vr.data_col in _INIT_COLS or k in _INIT_COLS else 0)
+             for k, vr in view_reqs.items()])
         self.buffers: Dict[str, List] = {}
         # The simple timestep count for this agent. Gets increased by one
         # each time a (non-initial!) observation is added.
         self.count = 0
 
-    def add_init_obs(self, episode_id: EpisodeID, agent_id: AgentID,
-                     env_id: EnvID, init_obs: TensorType,
+    def add_init_obs(self, episode_id: EpisodeID, agent_index: int,
+                     env_id: EnvID, t: int, init_obs: TensorType,
                      view_requirements: Dict[str, ViewRequirement]) -> None:
         """Adds an initial observation (after reset) to the Agent's trajectory.
```

Comment:
Change the description. It adds more than a single observation.

Reply:
No, it doesn't it's really just adds a single one. Same as it used to work w/ SampleBatchBuilder.